src/components/plot/selectors-grid.tsx

```diff
@@ -0,0 +1,104 @@
+import DropdownSelect from '@isrd-isi-edu/deriva-webapps/src/components/plot/dropdown-select';
+import { Option } from '@isrd-isi-edu/deriva-webapps/src/components/virtualized-select';
+import { Responsive, WidthProvider, ResponsiveProps as ResponsiveGridProps } from 'react-grid-layout';
+import '/node_modules/react-resizable/css/styles.css';
+import '/node_modules/react-grid-layout/css/styles.css';
+import { UserControlConfig, LayoutConfig } from '@isrd-isi-edu/deriva-webapps/src/models/plot';
+import { PlotTemplateParams } from '@isrd-isi-edu/deriva-webapps/src/hooks/chart';
+import { convertKeysSnakeToCamel } from '@isrd-isi-edu/deriva-webapps/src/utils/string';
+
+
+type UserControlsGridProps = {
+    /**
+     * selectors data to be rendered
+     */
+    userControlData: {
+        gridConfig: ResponsiveGridProps,
+        layout: LayoutConfig[],
+        userControlConfig: UserControlConfig[],
+        templateParams: PlotTemplateParams,
+    };
+    selectorOptions: Option[][];
+    setSelectorOptionChanged: any;
+    width: number | string;
+};
+
+//In simple cases a HOC WidthProvider can be used to automatically determine width upon initialization and window resize events.
+const ResponsiveGridLayout = WidthProvider(Responsive);
+
+
+/**
+ * It sets a new value in templateParams.$control_values based on selector, 
+ * triggers the setSelectorOptionChanged function, and returns the option
+ * @param option changed option
+ * @param userControlConfig configuration of the given selector
+ * @param templateParams 
+ * @param setSelectorOptionChanged setState method to indicate the change
+ * @returns 
+ */
+const handleChange = (option: Option, userControlConfig: UserControlConfig, templateParams: PlotTemplateParams, setSelectorOptionChanged: any) => {
+    if (option) {
+        setSelectorOptionChanged(true);
+        const uid = userControlConfig?.uid;
+        const valueKey = userControlConfig?.request_info?.value_key;
+        templateParams.$control_values[uid].values = {
+            [valueKey]: option.value, //else use default value
+        };
+        return option;
+    }
+};
+
+
+const SelectorsGrid = ({ userControlData, selectorOptions, setSelectorOptionChanged, width }: UserControlsGridProps): JSX.Element => {
```

Comment:
This should be change to `UserControlsGrid` and the file should be renamed to `user-controls-grid.tsx`. Make sure to update import statements